sdk/data/azcosmos/cosmos_client.go

```diff
@@ -75,10 +76,28 @@ func NewClient(endpoint string, cred azcore.TokenCredential, o *ClientOptions) (
 	if err != nil {
 		return nil, err
 	}
-	scope, err := createScopeFromEndpoint(endpointUrl)
-	if err != nil {
-		return nil, err
+
+	var scope []string
+	builtInAudience := cloud.AzurePublic.Services["cosmosDB"].Audience
+
+	if o != nil && o.ClientOptions.Cloud.Services != nil {
+		if svcCfg, ok := o.ClientOptions.Cloud.Services[cloud.ServiceName("cosmosDB")]; ok && svcCfg.Audience != "" {
+			if svcCfg.Audience == builtInAudience {
+				scope = []string{builtInAudience}
+				log.Write(azlog.EventRequest, fmt.Sprintf("Using built-in Cosmos DB audience: %s", builtInAudience))
+			} else {
+				log.Write(azlog.EventRequest, fmt.Sprintf("WARNING: Ignoring custom audience %q, using account scope instead", svcCfg.Audience))
```

Comment:
It's unclear why the custom value is being ignored. Isn't the idea to use the specified value?

Reply:
Updated the logic and now we use the audience value provided by the end user as part of client options. If end user did not set this value then used account based scope.